src/Components/Stats/Stats.js

```diff
@@ -1,38 +1,354 @@
-  );
+    <>
+      <h2 className={style.prompt}>Usage</h2>
+      {sessionLog.length ? 
+        <>
+          <div className={style.frequencyStatisticsContainer}>
+            <section className={style.pieChartContainer}>
+              <PieChart
+                data={
+                  [
+                    { title: 'Yoga', value: pieChartData.movementTotal, color: '#F4D2D0'},
+                    { title: 'Meditation', value: pieChartData.meditationTotal, color: '#7987A1' },
+                    { title: 'Somatics', value: pieChartData.somaticTotal, color: '#FFFFFF' },
+                  ]
+                }
+                style={{ height: '250px' }}
+                lineWidth={60}
+                labelStyle={{ fontSize: '12px', fontWeight: '600' }}
+                label={({ dataEntry }) => Math.round(dataEntry.percentage) + '%'}
+                labelPosition={68}
+                className={style.pieChart}
+              />
+              <section className={style.pieChartLegend}>
+                <div className={style.labelContainer}>
+                  <p className={style.legendLabel}>Somatic Exercises</p>
+                  <div className={style.somaticLabelSwatch} />
+                </div>
+                <div className={style.labelContainer}>
+                  <p className={style.legendLabel}>Yoga/Movement </p>
+                  <div className={style.movementLabelSwatch} />
+                </div>
+                <div className={style.labelContainer}>
+                  <p className={style.legendLabel}>Meditation/Breathwork</p>
+                  <div className={style.meditationLabelSwatch} />
+                </div>
+              </section>
+            </section>
+            <section className={style.frequencyWidgetsContainer}>
+              <h3 className={style.frequencyWidget}>
+                You have completed
+                <h2 
+                  className={style.frequencyStatisticValue} data-testid='sessionCount'
+                >
+                  {frequencyStatistics.sessionCount}
+                </h2>
+                sessions
+              </h3>
+              <h3 className={style.frequencyWidget}>
+                Average focus interval
+                <h2 
+                  className={style.frequencyStatisticValue} data-testid='focusIntervalAverage'
+                >
+                  {frequencyStatistics.averageFocusInterval.toFixed(2)}
+                </h2>
+                in minutes
+              </h3>
+              <h3 className={style.frequencyWidget}>
+                Average rest interval
+                <h2 
+                  className={style.frequencyStatisticValue} 
+                  data-testid='restIntervalAverage'
+                >
+                  {frequencyStatistics.averageRestInterval.toFixed(2)}
+                </h2>
+                in minutes
+              </h3>
+            </section>
+          </div>
+          <section className={style.moodWidgetsContainer}>
```

Comment:
Was it complicated to try to include conditional rendering for the mood widgets because it's possible that some sessions could have mood ratings while others didn't, based on when mood ratings were disabled? Meaning you would have to check that every single session in the log had mood ratings of 0 to know that these stats shouldn't be displayed? I totally see how that is complicated and not something we should worry about right now, but just wondering if I'm thinking about that correctly

Reply:
Yeah I'm not quite sure what to do with this! Do we want to only render these stats when mood is enabled? I was thinking about an instance in which a user tracked their mood for a while, but then turned off the ratings (or vice-versa). Though, in the `getAverageMood` method, I think we can use the `newCount` portion of the returned array to render only for categories of content with this value `> 1`! Will play around with this...

In the `getAverageMood` method, I included the line `if(session[moodRating])` as a way to filter out the zeroes- so these shouldn't be included for the average calculation!